pkg/proxy/proxy.go

```diff
@@ -134,9 +154,16 @@ func (p *SprayProxy) HandleProxy(c *gin.Context) {
 		newURL := copy.Request.URL
 		newURL.Host = backendURL.Host
 		newURL.Scheme = backendURL.Scheme
+
+		// if server post on non root endpoint e.g /proxy
+		// remove /proxy from the copied backend URL
+		if nonRootEndpoint {
+			newURL.Path = strings.TrimPrefix(newURL.Path, "/proxy")
```

Comment:
I'll suggest to move that to `HandleProxyEndpoint`. So instead of passing the `nonRootEndpoint` parameter to `handleProxyCommon`, we can simply alter the URL path there. So it'll become:
`func (p *SprayProxy) HandleProxyEndpoint(c *gin.Context) {
        // comment goes here
        c.Request.URL.Path = strings.TrimPrefix(c.Request.URL.Path, "/proxy")
        handleProxyCommon(p, c)
}`
That way we keep `handleProxyCommon` "common" and we skip that extra parameter.

Reply:
@enarha I have added the fix in [commit](https://github.com/redhat-appstudio/sprayproxy/pull/30/commits/7f2ddc624272ecfe239e51c6a3188b4a44b9a15d) . Let me know if everything looks good I will squash it.

Reply:
Yes. Please squash the commits and I'll lgtm.

Reply:
Done